File: pkg/java/src/main/java/dev/openfga/language/PartialConditionParamTypeRef.java
package dev.openfga.language;

import dev.openfga.sdk.api.model.ConditionParamTypeRef;
import dev.openfga.sdk.api.model.TypeName;

import java.util.ArrayList;
import java.util.List;

public class PartialConditionParamTypeRef {
    private TypeName typeName;
    private List<ConditionParamTypeRef> genericTypes = new ArrayList<>();

    public ConditionParamTypeRef asConditionParamTypeRef() {
        return new ConditionParamTypeRef()
                .typeName(typeName)
                .genericTypes(genericTypes);
    }

    public TypeName getTypeName() {
        return typeName;
    }

    public void setTypeName(TypeName typeName) {
        this.typeName = typeName;
    }

    public List<ConditionParamTypeRef> getGenericTypes() {
        return genericTypes;
    }

    public void setGenericTypes(List<ConditionParamTypeRef> genericTypes) {
        this.genericTypes = genericTypes;
    }
}

File: pkg/csharp/src/OpenFga.Language/PartialConditionParamTypeRef.cs
using OpenFga.Sdk.Model;

namespace OpenFga.Language;

/// <summary>
/// Represents a partial condition parameter type reference used during parsing.
/// This class is used internally during DSL parsing to build condition parameter type references.
/// </summary>
public class PartialConditionParamTypeRef {

    /// <summary>
    /// Gets or sets the type name for this parameter.
    /// </summary>
    public TypeName TypeName { get; set; }

    /// <summary>
    /// Gets or sets the generic types for this parameter (if applicable).
    /// </summary>
    public List<ConditionParamTypeRef>? GenericTypes { get; set; }

    /// <summary>
    /// Converts this partial type reference to a complete ConditionParamTypeRef.
    /// </summary>
    /// <returns>A complete ConditionParamTypeRef instance</returns>
    public ConditionParamTypeRef AsConditionParamTypeRef() {
        return new ConditionParamTypeRef() {
            TypeName = TypeName,
            GenericTypes = GenericTypes
        };
    }
}